docs/configuredgraphfactory.txt

```diff
@@ -0,0 +1,442 @@
+def g1 = ConfiguredGraphFactory.open("graph1");
+----
+
+3) Update a graph configuration that was created using a template configuration that has been updated:
+
+[source, gremlin]
+----
+Map map = new HashMap();
+map.put("storage.backend", "cassandrathrift");
+map.put("storage.hostname", "127.0.0.1");
+ConfigurationManagementGraph.getInstance().createTemplateConfiguration(new
+MapConfiguration(map));
+
+def g1 = ConfiguredGraphFactory.create("graph1");
+
+// Update template configuration
+Map map = new HashMap();
+map.put("index.search.backend", "elasticsearch");
+map.put("index.search.hostname", "127.0.0.1");
+map.put("index.search.elasticsearch.transport-scheme", "http");
+ConfigurationManagementGraph.getInstance().updateTemplateConfiguration(new
+MapConfiguration(map));
+
+// Remove Configuration
+ConfigurationManagementGraph.getInstance().removeConfiguration("graph1");
+
+// Close graph on all JanusGraph nodes
+ConfiguredGraphFactory.close("graph1");
+
+// Recreate
+ConfiguredGraphFactory.create("graph1");
+// Now this graph's configuration is guaranteed to be updated
+----
+
+[[janusgraphmanager]]
+=== JanusGraphManager
+
+The `JanusGraphManager` is a `Singleton` adhering to the TinkerPop graphManager specifications.
+
+In particular, the `JanusGraphManager` provides
+1) a coordinated mechanism by which to instantiate graph references on a given JanusGraph node
+2) a graph reference tracker (or cache)
+
+Any graph you create using the "graph.graphname" property will go through the `JanusGraphManager` and thus be instantiated in a coordinated fashion. The graph reference will also be placed in the graph cache on the JVM in question.
+
+Thus, any graph you open using the "graph.graphname" property that has already been instantiated on the JVM in question will be retrieved from the graph cache.
+
+This is why updates to your configurations require a few steps to guarantee correctness.
+
+[[usingthejanusgraphmanager]]
+==== How To Use The JanusGraphManager
+
+
+This is a new configuration option you can use when defining a property in your configuration that defines how to access a graph. All configurations that include this property will result in the graph instantiation happening through the `JanusGraphManager` (process explained above).
+
+For backwards compatability, any graphs that do not supply this parameter but supplied at server start in your graphs {} object in your .yaml file, these graphs will be bound through the JanusGraphManager denoted by their `key` supplied for that graph. For example, if your .yaml graphs object looks like:
+
+[source, properties]
+----
+graphs {
+graph1: conf/graph1.properties,
+graph2: conf/graph2.properties
+}
+----
+
+but `conf/graph1.properties` and `conf/graph2.properties` do not include the property `graph.graphname`, then these graphs will be stored in the JanusGraphManager and thus bound in your gremlin script executions as `graph1` and `graph2`, respectively.
```

Comment:
This line uses \`graph.graphname\` (i.e., backticks, no quotes) while other lines use "graph.graphname" (i.e., quotes, no backticks.

Please choose one and be consistent with it; since it's a properly that's used verbatim, it should probably be in backticks, no quotes.